realm/realm-library/src/objectServer/java/io/realm/mongodb/User.java

```diff
@@ -54,7 +56,7 @@
  * @see io.realm.mongodb.sync.SyncConfiguration.Builder#Builder(User, String)
  */
 @Beta
-public class User {
+public class User extends StreamNetworkTransport {
```

Comment:
I would avoid doing this given this class will be heavily modified soon due to the decoupling of the session info stuff.

The only thing the `StreamNetworkTransport` seems to need is `osUser.getNativePtr()` and a reference to the `app` to be able to call `app.makeStreamingRequest()` and `app.networkTransport.sendStreamingRequest`. Try to encapsulate it so that it can be done from a separate object.

Reply:
Alternatively, you could also pass a pointer to the native app and implement the `private static native OsJavaNetworkTransport.Request nativeMakeStreamingRequest(long nativeAppPtr, long nativeUserPtr, String functionName, String bsonArgs, String serviceName)` method for this new class and its corresponding native counterpart. Then it should be trivial to call `app->make_streaming_request` from c++.